### PR TITLE
fix(UserInput.js, Validation.js): fix #41 Create a form validation co…

### LINKED ISSUE #41: Create form validation component
The current form validation causes bad UI on some browsers (edge). Add an unified component to solve this. We can add more features to that validation component in the future.

### PR DESCRIPTION
**Pull request by Taiji-Yang**
…mponent

-The current form validation causes bad UI on some browsers (edge)
-They now have same validation check UI
-added Validation.js as the new form validation component
-changed UserInput.js to use this component
-used semantic-ui-react as UI

### CODE REVIEW

File: src/views/home/UserInput.js [hunk 1]
```diff
@@ -40,15 +44,25 @@ export default class UserInput extends Component {
     e.preventDefault();
     const body = JSON.stringify(this.state);
     const res = await createTicket(body);
-    if (!res) {
+    if (!res ) {
```
Comment (alejx) [inline src/views/home/UserInput.js]: Remove the extra space

File: src/views/home/UserInput.js [hunk 2]
```diff
@@ -40,15 +44,25 @@ export default class UserInput extends Component {
...
+  handlebut = () => {
+    if (this.state.url === '') {
+      this.setState({inputValid: false});
+    } else {
+      this.setState({inputValid: true});
+    }
+  }
+
   render() {
+    console.log(this.state.inputValid);
```
Comment (alejx) [inline src/views/home/UserInput.js]: Remove logs

File: src/views/home/Validation.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { Input, Button, Form, Header } from 'semantic-ui-react';
+
+export default class Validation extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+render() {
+  let result;
+    if(this.props.required && this.props.data === '' && !(this.props.inputValid)) {
```
Comment (alejx) [inline src/views/home/Validation.js]: It looks like a wrong indentation here. Did you run Prettier before committing?

File: src/views/home/UserInput.js [hunk 4]
```diff
@@ -40,15 +44,25 @@ export default class UserInput extends Component {
...
+    if (!res ) {
       console.error('Error creating ticket!!');
       // TODO: let user know about error
     } else {
       console.log(res);
       this.setState({ newTicket: res.ticket });
     }
   };
+
+  handlebut = () => {
```
Comment (alejx) [inline src/views/home/UserInput.js]: Please follow camelCase.

**Comment by Taiji-Yang** [comment]
Thanks for editing! I will fix that

**Pull request closed without merging**